glib/src/param_spec.rs

```diff
@@ -384,6 +384,98 @@ macro_rules! define_param_spec_numeric {
     };
 }
 
+macro_rules! define_builder {
+    (@constructors $rust_type:ident, $builder_type:ident $(($($req_ident:ident: $req_ty:ty,)*))?) => {
+        impl<'a> $builder_type<'a> {
+            pub fn new(name: &'a str, $($($req_ident: $req_ty)*)?) -> Self {
+                Self {
+                    name,
+                    $($($req_ident: Some($req_ident),)*)?
+                    ..Default::default()
+                }
+            }
+        }
+
+        impl $rust_type {
+            pub fn builder<'a>(name: &'a str, $($($req_ident: $req_ty),*)?) -> $builder_type<'a> {
+                $builder_type::new(name, $($($req_ident),*)?)
+            }
+        }
+    };
+    (
+        $rust_type:ident, $builder_type:ident {
+            $($field_id:ident: $field_ty:ty $(= $field_expr:expr)?,)*
+        }
+        $(requires $required_tt:tt)?
+    ) => {
+        #[derive(Default)]
+        pub struct $builder_type<'a> {
+            name: &'a str,
+            nick: Option<&'a str>,
+            blurb: Option<&'a str>,
+            flags: Option<crate::ParamFlags>, // `ParamFlags` doesn't implement `Default` so I wrap it in `Option`
+            $($field_id: Option<$field_ty>),*
+        }
+        impl<'a> $builder_type<'a> {
+            /// Default: `self.name`
+            #[must_use]
+            pub fn nick(mut self, nick: &'a str) -> Self {
+                self.nick = Some(nick);
+                self
+            }
+            /// Default: `self.name`
+            #[must_use]
+            pub fn blurb(mut self, blurb: &'a str) -> Self {
+                self.blurb = Some(blurb);
+                self
+            }
+
+            /// Default: `glib::ParamFlags::READWRITE`
+            #[must_use]
+            pub fn flags(mut self, flags: crate::ParamFlags) -> Self {
+                self.flags = Some(flags);
+                self
+            }
+
+            $(
+            $(#[doc = concat!("Default: `", stringify!($field_expr), "`")])?
+            #[must_use]
+            pub fn $field_id(mut self, value: $field_ty) -> Self {
+                self.$field_id = Some(value);
+                self
+            }
+            )*
+
+            pub fn build(self) -> ParamSpec {
```

Comment:
Should add a `#[must_use]` here too (what's the point of creating a pspec just to not use it? most likely a mistake)